src/custom/state/claim/hooks/index.ts

```diff
@@ -698,24 +728,61 @@ export function useClaimDispatchers() {
       // claim row selection
       setSelected: (payload: number[]) => dispatch(setSelected(payload)),
       setSelectedAll: (payload: boolean) => dispatch(setSelectedAll(payload)),
-    }
-    return dispatchers
-  }, [dispatch])
+    }),
+    [dispatch]
+  )
 }
 
 export function useClaimState() {
   return useSelector((state: AppState) => state.claim)
 }
 
 /**
- * Extend the Payable optional param
+ * Gets an array of available claims parsed and sorted for the UI
+ *
+ * Syntactic sugar on top of `useUserClaims`
+ *
+ * @param account
  */
-function _extendFinalArg(args: ClaimManyFnArgs, extendedArg: Record<any, any>) {
-  const lastArg = args.pop()
-  args.push({
-    ...lastArg, // add back whatever is already there
-    ...extendedArg,
-  })
+export function useUserEnhancedClaimData(account: Account): EnhancedUserClaimData[] {
+  const { available } = useClassifiedUserClaims(account)
+  const { chainId: preCheckChainId } = useActiveWeb3React()
+
+  const checkType = useCallback((type) => Number(FREE_CLAIM_TYPES.includes(type)), [])
```

Comment:
Note to self: I see no reason for this fn to be inside the hook. We could potentially extract the sort fn as well